File: library/src/main/java/com/pengrad/telegrambot/model/request/InputMediaDocument.java
package com.pengrad.telegrambot.model.request;

import com.pengrad.telegrambot.request.ContentTypes;

import java.io.File;
import java.io.Serializable;

/**
 * Stas Parshin
 * 28 July 2018
 */
public class InputMediaDocument extends InputMedia<InputMediaDocument> implements Serializable {
    private final static long serialVersionUID = 0L;

    private Boolean disable_content_type_detection;

    public InputMediaDocument(String media) {
        super("document", media);
    }

    public InputMediaDocument(File media) {
        super("document", media);
    }

    public InputMediaDocument(byte[] media) {
        super("document", media);
    }

    public InputMediaDocument disableContentTypeDetection(boolean disableContentTypeDetection) {
        disable_content_type_detection = disableContentTypeDetection;
        return this;
    }

    @Override
    public String getDefaultFileName() {
        return ContentTypes.DOC_FILE_NAME;
    }

    @Override
    public String getContentType() {
        return ContentTypes.DOC_MIME_TYPE;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InputMediaDocument.kt
package com.pengrad.telegrambot.model.request

import com.pengrad.telegrambot.request.ContentTypes

import java.io.File
import java.io.Serializable

/**
 * Stas Parshin
 * 28 July 2018
 */
class InputMediaDocument : InputMedia<InputMediaDocument>, Serializable {

    override val defaultFileName: String
        get() = ContentTypes.DOC_FILE_NAME

    override val contentType: String
        get() = ContentTypes.DOC_MIME_TYPE

    constructor(media: String) : super("document", media) {}

    constructor(media: File) : super("document", media) {}

    constructor(media: ByteArray) : super("document", media) {}

    companion object {
        private const val serialVersionUID = 0L
    }
}
